### PR TITLE
feat(LERN-102): implemented getGradeDetails endpoint

### PR DESCRIPTION
**Pull request by A1eeee**
Also added column subjectId to tblUser and changed renaming in the frontend.

### CODE REVIEW

File: backend/src/main/resources/db/changelog/changes/2024-06-29_column_subjectId_tblUser.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                   https://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
+
+    <changeSet id="2024-06-29_column_subjectId_tblUser" author="alejandro.wenzler">
+        <addColumn tableName="t_user">
+            <column name="subject_id" type="uuid">
+                <constraints nullable="true"/>
+            </column>
+        </addColumn>
```
Comment (deepspace1000) [inline backend/src/main/resources/db/changelog/changes/2024-06-29_column_subjectId_tblUser.xml:12]: do it the other way around. add a column teacher_id to t_subject.

**Review by deepspace1000 — CHANGES_REQUESTED** [review]

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ custom/admin-landing-page*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

File: backend/src/main/kotlin/ch/learntrack/backend/persistence/SubjectQueries.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import ch.learntrack.backend.persistence.tables.daos.SubjectDao
+import ch.learntrack.backend.persistence.tables.pojos.Subject
+import ch.learntrack.backend.persistence.tables.references.GRADE
+import ch.learntrack.backend.persistence.tables.references.SUBJECT
+import java.util.UUID
+
+public fun SubjectDao.fetchSubjectsByGradeId(gradeId: UUID): List<Subject> = ctx()
+    .select()
+    .from(GRADE)
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/persistence/SubjectQueries.kt:11]: when you are going from SubjectDao please use .from(SUBJECT)

File: backend/src/main/kotlin/ch/learntrack/backend/subject/SubjectDto.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package ch.learntrack.backend.subject
+
+import java.util.UUID
+
+public data class SubjectDto(
+    val id: UUID,
+    val name: String,
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/subject/SubjectDto.kt:7]: shouldn't this also have gradeId and teacherId?

Comment (deepspace1000) [reply backend/src/main/kotlin/ch/learntrack/backend/subject/SubjectDto.kt:7]: When you add the teacherId you can call in the adminGradeService the get userById function withouth the need to write a query to join user on subject. you can call the default jooy query UserDao.fetchOneByid(teacherId).

File: backend/src/main/kotlin/ch/learntrack/backend/user/UserService.kt [hunk 4]
```diff
@@ -50,6 +51,9 @@ public class UserService(
     public fun searchForUserByEmail(email: String, userRole: UserRole): List<User> =
         userDao.searchUserByUserRoleAndEmailSearchQuery(email, userRole)
 
+    public fun getUserBySubjectId(subjectId: UUID): UserDto? =
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/user/UserService.kt:54]: should not be needed anymore

File: backend/src/main/kotlin/ch/learntrack/backend/persistence/UserQueries.kt [hunk 5]
```diff
@@ -34,3 +35,11 @@ public fun UserDao.searchUserByUserRoleAndEmailSearchQuery(email: String, userRo
...
     .fetch()
     .into(User::class.java)
+
+public fun UserDao.fetchTeacherBySubjectId(subjectId: UUID): User? = ctx()
+    .select()
+    .from(USER)
+    .join(SUBJECT)
+    .on(USER.ID.eq(SUBJECT.TEACHER_ID))
+    .where(SUBJECT.ID.eq(subjectId))
+    .fetchOneInto(User::class.java)
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/persistence/UserQueries.kt:45]: should not be needed anymore

**Review by deepspace1000 — COMMENTED** [review]
i think we can remove some code...

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ custom/admin-landing-page*: 


| Before | After |
| --- | --- |
|  |  |

**Review by deepspace1000 — CHANGES_REQUESTED** [review]
Split the Dto up into 2 different apis.

**Pull request closed without merging**